tests/attach_readmem.rs

```diff
@@ -0,0 +1,57 @@
+//! This is a simple test to attach to already running debugee process
+
+use nix::{
+    sys::wait::waitpid,
+    unistd::{execv, fork, ForkResult, Pid},
+};
+use std::ffi::CString;
+
+mod test_utils;
+
+#[cfg(target_os = "linux")]
+use headcrab::{symbol::Dwarf, target::LinuxTarget, target::UnixTarget};
+
+static BIN_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/testees/hello");
+
+// Ignoring because most linux distributions have attaching to a running process disabled
+#[ignore]
+#[cfg(target_os = "linux")]
+#[test]
+fn attach_readmem() -> Result<(), Box<dyn std::error::Error>> {
+    test_utils::ensure_testees();
+
+    let debuginfo = Dwarf::new(BIN_PATH)?;
+
+    let str_addr = debuginfo
+        .get_var_address("STATICVAR")
+        .expect("Expected static var has not been found in the target binary");
+
+    match fork()? {
+        ForkResult::Parent { child, .. } => {
+            let target = LinuxTarget::attach(child)?;
```

Comment:
There is a race condition here. The child may already have exited when you try to attach. Maybe add another testee that has a busy loop and then wait like 10ms before attaching and at the end kill the testee.